tests/flytekit/unit/models/test_types.py

```diff
@@ -81,6 +83,16 @@ def test_literal_types():
     assert obj == _types.LiteralType.from_flyte_idl(obj.to_flyte_idl())
 
 
+def test_annotated_literal_types():
```

Comment:
Can you add a few tests for more complex literal types as well?

Reply:
added for:
* enum
* dict
* dataclass